pytorch_blade/pytorch_blade/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscConvertTorchToDiscMhlo.cpp

```diff
@@ -123,12 +129,137 @@ class ConvertOperatorOp : public OpConversionPattern<OperatorOp> {
+          scale,
+          castedZeroPoint,
+          useSymmetricAttr,
+          axisAttr,
+          useDynamicAttr,
+          roundModeAttr);
+    } else {
+      return op.emitError("Unsupported kind of torch.operator.");
+    }
+    rewriter.replaceOp(op, newOutput->getResult(0));
+    return success();
+  }
+
+  LogicalResult convertCustomCallOp(
+      OperatorOp op,
+      OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const {
+    auto ctx = rewriter.getContext();
+    Location loc = op.getLoc();
+    auto operands = adaptor.operands();
+    SmallVector<Type> resultTypes;
+    for (const auto& result : op.getResults()) {
+      auto torchMlirResultTy = result.getType().dyn_cast<ValueTensorType>();
+      auto resultTy = getTypeConverter()
+                          ->convertType(torchMlirResultTy)
+                          .dyn_cast<RankedTensorType>();
+      if (!resultTy) {
+        return op.emitError("failed to get type of output.");
+      }
+      resultTypes.push_back(resultTy);
+    }
+
+    const std::vector<std::string> requiredAttrName{
+        "call_target_name",
+        "device",
+        "input_placements",
+        "output_placements",
+        "input_layouts",
+        "output_layouts",
+        "expected_input_layouts",
+        "expected_output_layouts"};
 
-      return success();
+    for (const auto& n : requiredAttrName) {
+      if (!op->hasAttr(n)) {
+        return op.emitError()
+            << n
+            << " attribute should be provided for torch_blade.custom_call.";
+      }
+    }
+
+    StringAttr callTarget =
+        op->getAttr("call_target_name").dyn_cast<StringAttr>();
+    StringAttr device = op->getAttr("device").dyn_cast<StringAttr>();
+    StringAttr inputPlacements =
+        op->getAttr("input_placements").dyn_cast<StringAttr>();
+    StringAttr outputPlacements =
+        op->getAttr("output_placements").dyn_cast<StringAttr>();
+    StringAttr inputLayouts =
+        op->getAttr("input_layouts").dyn_cast<StringAttr>();
+    StringAttr outputLayouts =
+        op->getAttr("output_layouts").dyn_cast<StringAttr>();
+    StringAttr expectedInputLayouts =
+        op->getAttr("expected_input_layouts").dyn_cast<StringAttr>();
+    StringAttr expectedOutputLayouts =
+        op->getAttr("expected_output_layouts").dyn_cast<StringAttr>();
+
+    DictionaryAttr customAttrs = DictionaryAttr::get(ctx, {});
+    if (op->hasAttr("custom_attrs")) {
+      customAttrs = op->getAttr("custom_attrs").dyn_cast<DictionaryAttr>();
+    }
+    Operation* newOutput = rewriter.create<mhlo_disc::CustomCallV2Op>(
+        loc,
+        TypeRange(resultTypes),
+        operands,
+        callTarget,
+        customAttrs,
+        false, // has_side_effect
+        device,
+        inputPlacements,
+        outputPlacements,
+        inputLayouts,
+        outputLayouts,
+        expectedInputLayouts,
+        expectedOutputLayouts);
+    rewriter.replaceOp(op, newOutput->getResult(0));
```

Comment:
using `newOutput->getResults()` in case there are multiple outputs. And better to add a UT for it.

Reply:
done, and have added a ut for multi outputs